### PR TITLE
various session improvements

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -716,11 +710,9 @@ func (s *session) logPacket(packet *packedPacket) {
 		// We don't need to allocate the slices for calling the format functions
 		return
 	}
-	if utils.Debug() {
-		utils.Debugf("-> Sending packet 0x%x (%d bytes) for connection %x, %s", packet.number, len(packet.raw), s.connectionID, packet.encryptionLevel)
-		for _, frame := range packet.frames {
-			frames.LogFrame(frame, true)
-		}
+	utils.Debugf("-> Sending packet 0x%x (%d bytes) for connection %x, %s", packet.number, len(packet.raw), s.connectionID, packet.encryptionLevel)
```
Comment (lucas-clemente) [inline session.go]: This change will cause a lot of unnecessary slice allocations if debug logging is not enabled.

Comment (marten-seemann) [inline session.go]: It won't, because there's a `return` 2 lines above.

Comment (lucas-clemente) [inline session.go]: Huh, okay. Why did we have this if then at all…

Comment (lucas-clemente) [inline session.go]: Also, GH should really show more in the diff view 🙄

**Pull request merged**